### PR TITLE
Add a back button to search, Make component

### LINKED ISSUE #994: Getting stuck in "Search"
If you search for something from your project's talk page (e.g. https://www.zooniverse.org/#/projects/aliburchard/wildebeest-watch/talk)  and get "no results" then you get stuck with nothing to click on to "clear" the search. Even if you refresh the page or clear the text out of the search box, you still just see "no results found."

You *can* click on project "Talk" but it seems sort of round about and unintuitive instead of being able to clear the search filters.

### PR DESCRIPTION
**Pull request by aweiksnar**
- Closes #994

### CODE REVIEW

**Comment by parrish** [comment]
Looks like the conflicts on this are just the react-router requires

**Comment by aweiksnar** [comment]
Yeah I started this branch before that was merged. Just rebased with master

File: app/talk/search.cjsx [hunk 1]
```diff
@@ -69,6 +69,10 @@ module.exports = React.createClass
     numberOfResults = @state.results.length
 
     <div className="talk-search">
+      <button  className="link-style" type="button" onClick={-> window.history.back()}>
```
Comment (chrissnyder) [inline app/talk/search.cjsx]: Could this use https://github.com/rackt/react-router/blob/0.13.x/doc/06%20Mixins/Navigation.md#goback instead?

**Comment by chrissnyder** [comment]
Looks good!

File: app/talk/search.cjsx (hunk 1 above)
Comment (aweiksnar) [inline app/talk/search.cjsx]: Yep, done

**Comment by chrissnyder** [comment]
:shipit:

**Pull request merged**